### PR TITLE
[FLINK-31100][Stream] Support the short topic pattern subscriber on default tenant and namespace.

### PR DESCRIPTION
**Pull request by syhily**
## Purpose of the change

The connector can subscribe the topics by using regular expression. But it can't use any simple regex like `some.*`. Any regex should in `tenant/namespace/regex` format. Since Pulsar support simple topic name which live under `public` tenant and `default` namespace. I think it's awesome to support same ability in regex subscription.

## Brief change log

- Change the internal design of `TopicPatternSubscriber`. The regex will be convert into a full regex by using `TopicName`.

There is [a known bug](https://github.com/apache/pulsar/issues/19316) in Pulsar 2.11.0. We can't subscribe the non-persistent topics by using regex now. We also create [a JIRA ticket](https://issues.apache.org/jira/browse/FLINK-31107) for tracking this bug.

## Verifying this change

This change added tests and can be verified as follows:

- `PulsarSubscriberTest.simpleTopicPatternSubscriber`
- `PulsarSubscriberTest.topicPatternSubscriber`
- `PulsarSubscriberTest.subscribeNonPartitionedTopicPattern`

## Significant changes

- [ ] Dependencies have been added or upgraded
- [ ] Public API has been changed (Public API is any class annotated with `@Public(Evolving)`)
- [ ] Serializers have been changed
- [x] New feature has been introduced
    - If yes, how is this documented? (docs)

### CODE REVIEW

File: flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/testutils/runtime/remote/PulsarRemoteRuntime.java [hunk 1]
```diff
@@ -28,8 +28,12 @@ public PulsarRemoteRuntime(String serviceUrl, String adminUrl) {
 
     @Override
     public PulsarRuntime withConfigs(Map<String, String> configs) {
-        throw new UnsupportedOperationException(
-                "We can't change the broker configs on a running instance.");
+        if (!configs.isEmpty()) {
```
Comment (reswqa) [inline flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/testutils/runtime/remote/PulsarRemoteRuntime.java:31]: What is the purpose of this change? In what scenario will we pass in an empty Map.

Comment (syhily) [reply flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/testutils/runtime/remote/PulsarRemoteRuntime.java:31]: This is a local debug class for investigate the root cause from Pulsar side. We may use it with `PulsarTestSuiteBase`, it will pass an empty map by default.

Comment (reswqa) [reply flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/testutils/runtime/remote/PulsarRemoteRuntime.java:31]: Ok

**Review by reswqa — APPROVED** [review]
Thanks @syhily, LGTM!
